rig/machine_control/machine_controller.py

```diff
@@ -1375,12 +1389,62 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
         self._x = x
         self._y = y
         self._machine_controller = machine_controller
+
+        # Store and clip the addresses
         self._start_address = start_address
-        self._end_address = end_address
+        self._end_address = max(start_address, end_address)
 
         # Current offset from start address
         self._offset = 0
 
+    def __getitem__(self, sl):
+        """Get a new file-like view of SDRAM covering the range indicated by
```

Comment:
It would be worth explicitly documenting what the offset is after splitting. I'm not sure which is better out of "always 0" or "the old offset shifted and clamped to the new range". I can't see the latter being too useful, except possibly if you are truncating something... but that seems a little dubious. Either way, it should be documented.